EffectEngine.cpp

```diff
@@ -147,9 +156,22 @@ bool EffectEngine::isValidEffect(const String effectName) {
 }
 
 void EffectEngine::setPixel(uint16_t idx, CRGB color) {
-    _ledDriver->setRValue(idx, (uint8_t)(color.r * _effectBrightness) );
-    _ledDriver->setGValue(idx, (uint8_t)(color.g * _effectBrightness) );
-    _ledDriver->setBValue(idx, (uint8_t)(color.b * _effectBrightness) );
+    if (_ledDriver->hasWhite()) {
+        _ledDriver->setRValue(idx, (uint8_t)(color.r * _effectBrightness) );
+        _ledDriver->setGValue(idx, (uint8_t)(color.g * _effectBrightness) );
+        _ledDriver->setBValue(idx, (uint8_t)(color.b * _effectBrightness) );
+    }
+    else {
+        _ledDriver->setValue(3 * idx + 0, (uint8_t)(color.r * _effectBrightness) );
```

Comment:
Can you help me understand why you are not always using `->set[RGB]]Value` function here?  It looks as though you've hard-coded the R/G/B ordering for non-white chipsets?

Reply:
so I think the ordering is translated in a later stage, so RGB must be hardcoded here..

I don't have a good answer for why I didnt just always use `->set[RGB]Value` except that I was trying to only add functionality for white without potentially adversely affecting existing functionality. Doing it the way I did here avoided the new dependency I added here on `->color_count` for non-white.

Looking at it closely there, it seems like there would be no harm to change the function to just be this:
```
        _ledDriver->setRValue(idx, (uint8_t)(color.r * _effectBrightness) );
        _ledDriver->setGValue(idx, (uint8_t)(color.g * _effectBrightness) );
        _ledDriver->setBValue(idx, (uint8_t)(color.b * _effectBrightness) );
```

Reply:
Yes, having these this would be good, as it would encapsulate the offset calculation for each color into its own function.  Having a single set of functions that abstract both color order and `color_count` is a great example of how to apply the single responsibility principle, and thus will help avoid bugs later.  Nice thinking!